src/System.Management.Automation/FormatAndOutput/common/ComplexWriter.cs

```diff
@@ -353,27 +355,58 @@ static StringManipulationHelper()
         private static IEnumerable<GetWordsResult> GetWords(string s)
         {
             StringBuilder sb = new StringBuilder();
-            GetWordsResult result = new GetWordsResult();
+            StringBuilder vtSeqs = null;
+            Dictionary<int, int> vtRanges = null;
 
+            var valueStrDec = new ValueStringDecorated(s);
+            if (valueStrDec.IsDecorated)
```

Comment:
Nit: The use of IsDecorated seems unwarranted. Rather, it should be an internal implementation of ValueStringDecorated.

Reply:
> (In general, if now PowerShell Formatting System is based on VT sequences all string sent to it should be ValueStringDecorated.)

I don't think that is necessary, or, I should say I don't see the benefit from doing that.

Reply:
> I think we can discuss about your idea in #16643

Is it right link? I don't see my comment there.
And yes, my proposal was to move all formatting (and path case normalization) codes to formatting system.

Reply:
I left a comment there about this topic: https://github.com/PowerShell/PowerShell/issues/16643#issuecomment-996984580
But it may be a better idea to open a new issue just about that, if one doesn't exist yet.